File: langgraph/result/qwen/7b-eq2js/49.js
function calculateModularExponentiation(n, p) {
    let ret = 1;
    let i = 0;
    while (i < n) {
        ret = (2 * ret) % p;
        i += 1;
    }
    return ret;
}

// function name: calculateModularExponentiation
// parameter names: n (base_count), p (modulus)
// variable names: ret (result), i (index)
// changed for loop to while loop

File: langgraph/data/semantic_equiv/49.py
def calculate_modular_exponentiation(number: int, prime: int) -> int:
    result = 1
    index = 0
    while index < number:
        result = (2 * result) % prime
        index += 1
    return result

# method name: modp -> calculate_modular_exponentiation
# parameter name: n -> number, p -> prime
# variable name: ret -> result, i -> index
# changed for loop to while loop